projects/js-packages/publicize-components/src/components/add-connection-modal/index.tsx

```diff
@@ -0,0 +1,81 @@
+import { Button, useBreakpointMatch } from '@automattic/jetpack-components';
+import { Modal } from '@wordpress/components';
+import { __, sprintf } from '@wordpress/i18n';
+import { Icon, chevronRight } from '@wordpress/icons';
+import classNames from 'classnames';
+import { useCallback, useState } from 'react';
+import { ConnectPage } from './connect-page/connect-page';
+import { getSupportedConnections } from './constants';
+import styles from './style.module.scss';
+
+const AddConnectionModal = ( { onCloseModal } ) => {
+	const [ currentService, setCurrentService ] = useState( null );
+
+	const [ isSmall ] = useBreakpointMatch( 'sm' );
+
+	const onServiceSelected = useCallback(
+		service => () => {
+			setCurrentService( service );
+		},
+		[]
+	);
+
+	const onBackClicked = useCallback( () => {
+		setCurrentService( null );
+	}, [] );
+
+	return (
+		<Modal
+			className={ classNames( styles.modal, {
+				[ styles[ 'service-selector' ] ]: ! currentService,
+				[ styles.small ]: isSmall,
+			} ) }
+			onRequestClose={ onCloseModal }
+			title={
+				currentService
+					? sprintf(
+							// translators: %s: Name of the service the user connects to.
+							__( 'Connecting a new %s account', 'jetpack' ),
+							currentService.title
+					  )
+					: __( 'Add a new connection to Jetpack Social', 'jetpack' )
+			}
+		>
+			{ currentService ? (
+				<ConnectPage service={ currentService } onBackClicked={ onBackClicked } />
+			) : (
+				<table>
+					<thead>
+						<tr>
+							<th></th>
+							<th></th>
+							<th></th>
+						</tr>
```

Comment:
Empty table headers are not good for a11y. May be we should use `grid` instead of table?

Reply:
This doesn't need to happen in this PR. We can work on it once designs are updated. This is fine for now.